modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpConnectionManager.java

```
@@ -324,5 +324,14 @@ private void populateSenderConfigurationOptions(SenderConfiguration senderConfig
         }
         senderConfiguration.setFollowRedirect(followRedirect == 1 ? true : false);
         senderConfiguration.setMaxRedirectCount(maxRedirectCount);
+
+        long idleTimeout = options.getIntField(Constants.IDLE_TIMEOUT_STRUCT_INDEX);
+        if (idleTimeout < 0 || (int) idleTimeout != idleTimeout) {
+            PrintStream out = System.out;
```

Better to throw an error in this case.